storm-core/src/jvm/org/apache/storm/Thrift.java

```diff
@@ -0,0 +1,351 @@
+    }
+
+    public static StormTopology._Fields[] getSpoutFields() {
+        return SPOUT_FIELDS;
+    }
+
+    public static class SpoutDetails {
+        private IRichSpout spout;
+        private Integer parallelism;
+        private Map conf;
+
+        public SpoutDetails(IRichSpout spout, Integer parallelism, Map conf) {
+            this.spout = spout;
+            this.parallelism = parallelism;
+            this.conf = conf;
+        }
+
+        public IRichSpout getSpout() {
+            return spout;
+        }
+
+        public Integer getParallelism() {
+            return parallelism;
+        }
+
+        public Map getConf() {
+            return conf;
+        }
+    }
+
+    public static class BoltDetails {
+        private Object bolt;
+        private Map conf;
+        private Integer parallelism;
+        private Map<GlobalStreamId, Grouping> inputs;
+
+        public BoltDetails(Object bolt, Map conf, Integer parallelism,
+                           Map<GlobalStreamId, Grouping> inputs) {
+            this.bolt = bolt;
+            this.conf = conf;
+            this.parallelism = parallelism;
+            this.inputs = inputs;
+        }
+
+        public Object getBolt() {
+            return bolt;
+        }
+
+        public Map getConf() {
+            return conf;
+        }
+
+        public Map<GlobalStreamId, Grouping> getInputs() {
+            return inputs;
+        }
+
+        public Integer getParallelism() {
+            return parallelism;
+        }
+    }
+
+    public static StreamInfo directOutputFields(List<String> fields) {
+        return new StreamInfo(fields, true);
+    }
+
+    public static StreamInfo outputFields(List<String> fields) {
+        return new StreamInfo(fields, false);
+    }
+
+    public static Grouping prepareShuffleGrouping() {
+        return Grouping.shuffle(new NullStruct());
+    }
+
+    public static Grouping prepareLocalOrShuffleGrouping() {
+        return Grouping.shuffle(new NullStruct());
+    }
+
+    public static Grouping prepareFieldsGrouping(List<String> fields) {
+        return Grouping.fields(fields);
+    }
+
+    public static Grouping prepareGlobalGrouping() {
+        return prepareFieldsGrouping(new ArrayList<String>()); // Is this array list
+    }
+
+    public static Grouping prepareDirectGrouping() {
+        return Grouping.direct(new NullStruct());
+    }
+
+    public static Grouping prepareAllGrouping() {
+        return Grouping.all(new NullStruct());
+    }
+
+    public static Grouping prepareNoneGrouping() {
+        return Grouping.none(new NullStruct());
+    }
+
+    public static Grouping prepareCustomStreamGrouping(Object obj) {
+        return Grouping.custom_serialized(Utils.javaSerialize(obj));
+    }
+
+    public static Grouping prepareCustomJavaObjectGrouping(JavaObject obj) {
+        return Grouping.custom_object(obj);
+    }
+
+    public static CustomStreamGrouping instantiateJavaObject(JavaObject obj) {
```

Comment:
Can we make this return an Object for now?  The case can happen in the code that uses it.